pandas/core/ops/array_ops.py

```diff
@@ -246,7 +259,10 @@ def comparison_op(left: ArrayLike, right: Any, op) -> ArrayLike:
                 "Lengths must match to compare", lvalues.shape, rvalues.shape
             )
 
-    if should_extension_dispatch(lvalues, rvalues):
+    if should_extension_dispatch(lvalues, rvalues) or (
```

Comment:
this is the same check as above, (L212) can you put the common parts in a function

Reply:
It's *slightly* different: here I need an additional `and not is_object_dtype(lvalues.dtype)`, because in that case we need to take the `comp_method_OBJECT_ARRAY` code path below

Reply:
sure i see that, that's why i said common parts (meaning the datetimes like + NaT)

Reply:
Yes, but check the brackets: currently the checks for the scalars is first combined with `is_object_dtype`, before being combined with `should_extension_dispatch`. So that means I cannot move those scalar checks inside `should_extension_dispatch` without changing the behaviour of the check.